src/ir/context.rs

```diff
@@ -273,7 +283,6 @@ impl<'ctx> BindgenContext<'ctx> {
             currently_parsed_types: vec![],
             parsed_macros: Default::default(),
             replacements: Default::default(),
-            effective_target: effective_target.unwrap(),
```

Comment:
I'd rather not. Even if we want something like this, we probably should store it in a more structured way rather than just storing a string, because string operations tend to be slower, and spreading that across the code could make things more fragile and probably hard to audit.

I would suggest we just add more flags into BindgenContext when needed, and do all the target detection at this single place.